networkx/algorithms/tests/test_dominance.py

```diff
@@ -39,19 +39,21 @@ def test_unreachable(self):
         }
 
     def test_irreducible1(self):
-        # Graph taken from Figure 2 of
-        # K. D. Cooper, T. J. Harvey, and K. Kennedy.
-        # A simple, fast dominance algorithm.
-        # Software Practice & Experience, 4:110, 2001.
+        """
+        .. [1]  Cooper, Keith D., Harvey, Timothy J. and Kennedy, Ken.
+                "A simple, fast dominance algorithm." (2006).
+                https://hdl.handle.net/1911/96345
+        """
```

Comment:
Let's keep the info on figure provenance. No need to worry about formatting here as the test suite is not included in the docs build:

```suggestion
        """
        Graph taken from figure 2 of "A simple, fast dominance algorithm." (2006).
        https://hdl.handle.net/1911/96345
        """
```